### PR TITLE
MS 48 : ADMIN api rest

### PR DESCRIPTION
**Pull request by amacazzaga**
• Add new model
• Add endpoints + controllers using GET, POST & PUT 
• Import Postman collection
• Validate data using mongoose schemas validation

### CODE REVIEW

File: src/models/admins.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    required: true,
+  },
+  username: {
+    type: String,
+    maxlength: 50,
+    required: true,
+  },
+  password: {
+    type: String,
+    maxlength: 20,
```
Comment (nicolasblasco) [inline src/models/admins.js:21]: @amacazzaga we should write this property type in camelCase --> maxLength

File: src/controllers/admins.js [hunk 2]
```diff
@@ -1,57 +1,56 @@
...
-    res.json(admin);
-  } else {
-    res.status(400).send("Admin not found");
-  }
+  Admins.findById(req.params.id)
+    .then((admin) => {
+      return res.status(200).json(admin);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/admins.js:19]: @amacazzaga we should change this HTTP response status code to 404: Not Found.

File: src/controllers/admins.js [hunk 3]
```diff
@@ -1,57 +1,56 @@
...
+const Admins = require("../models/admins");
 
 const getAll = (req, res) => {
-  res.json(admins);
+  Admins.find()
+    .then((admins) => {
+      return res.status(200).json(admins);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/admins.js:9]: @amacazzaga we should change this HTTP response status code to 404: Not Found.

File: src/controllers/admins.js [hunk 4]
```diff
@@ -1,57 +1,56 @@
...
-    res.json(admin);
-  } else {
-    res.status(400).send("Admin not found");
-  }
+  Admins.find({ name: req.params.name })
+    .then((admin) => {
+      return res.status(200).json(admin);
+    })
+    .catch((err) => {
+      return res.status(400).json(err);
```
Comment (nicolasblasco) [inline src/controllers/admins.js:28]: @amacazzaga we should change this HTTP response status code to 404: Not Found.

File: src/models/admins.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  email: {
+    type: String,
+    maxLength: 20,
+    required: true,
+  },
+  username: {
+    type: String,
+    maxLength: 50,
+    required: true,
```
Comment (nicolasblasco) [inline src/models/admins.js:17]: @amacazzaga we should add the unique type to this property so the username can't be repeated. Like this --> unique: true

File: src/models/admins.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+const AdminsSchema = new Schema({
+  name: {
+    type: String,
+    maxLength: 50,
+    required: true,
+  },
+  email: {
+    type: String,
+    maxLength: 20,
+    required: true,
```
Comment (nicolasblasco) [inline src/models/admins.js:12]: @amacazzaga we should add the unique type to this property so the email can't be repeated. Like this --> unique: true

**Review by JamirMansilla — CHANGES_REQUESTED** [review]
Agus resolve the conflicts and thats it 🛩️

**Review by damianalvarez — COMMENTED** [review]
@amacazzaga don't forget to use the correct format in the PR title and the commits

**Review by damianalvarez — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**